### PR TITLE
Corrective vel factor

### PR DESCRIPTION
**Pull request by mc4117**
As part of my conservative tracer 2d I have created this separate pull request where I have changed the implementation of the corrective velocity factor (see previous pull request https://github.com/thetisproject/thetis/pull/164). This is because I realised that the correction factor also needed to be applied to the boundary conditions which made a difference in the case of the conservative tracer.

### CODE REVIEW

File: thetis/options.py [hunk 1]
```diff
@@ -543,8 +543,10 @@ class ModelOptions2d(CommonModelOptions):
 
         Prints deviation from the initial mass to stdout.
         """).tag(config=True)
-    tracer_advective_velocity = FiredrakeVectorExpression(None, allow_none=True,
-                                                          help="Custom function to be used for the velocity variable in tracer advection equation").tag(config=True)
+    tracer_advective_velocity_factor = FiredrakeScalarExpression(
+        Constant(1.0), help="""
+        Custom factor which is multiplied to the velocity variable in tracer advection equation
+        """).tag(config=True)
```
Comment (jwallwork23) [inline thetis/options.py]: Maybe you could include a sentence on when it might be useful to use this parameter?

**Comment by tkarna** [comment]
Looking at the Huybrechts et al. (2010) paper, I guess you are defining the correction factor in the application py file? And this correction factor has been derived for sediments. i.e. it's not for any generic tracer.

**Comment by mc4117** [comment]
@jwallwork23  Yes that's correct. It's basically just changing a previous pull request (https://github.com/thetisproject/thetis/pull/164)
@tkarna Yes that's also correct. Unfortunately there isn't just a default corrective factor that you can use for all tracer applications and even for sediment transport it's dependent on all sorts of variables.

**Comment by jwallwork23** [comment]
I don't know a lot about tracer parametrisations. However, following from my previous observation, maybe we could just have both? i.e. `uv = alpha*uv + beta`? With defaults `alpha=1`, `beta=0`. Just an idea.

File: thetis/options.py (hunk 1 above)
Comment (tkarna) [reply thetis/options.py]: If the help string is long it's a good idea to split it into a short description + new line  + long description. E.g., in here https://github.com/thetisproject/thetis/blob/master/thetis/options.py#L405

Comment (mc4117) [reply thetis/options.py]: I have updated this with your suggestion

**Comment by mc4117** [comment]
I am happy to include it if necessary but I think a factor like the beta one you're talking about would probably normally go into something like the source term. I am also rewriting how thetis does source terms in my next pull request so I think it might be best to leave this for now?

**Comment by jwallwork23** [comment]
@mc4117 Fair enough, sounds good :)

**Pull request merged**